### ISSUE TITLE
[TCIM][Step 1][E] 接通 raw-GGUF static-span storage/H2D/residency

### ISSUE THREAD

**Issue by kkongling** [fixedpointworks/ds4#7]
Part of #2; governed by #1 and ADR-0001.

## Scope

- 复用 shared `ds4.c` 的 model fd/open/mmap、GGUF validation、span construction 与 readahead orchestration。
- 在 TCIM backend 实现 static 路径所需的 model fd/map、range/spans、storage/H2D、cache/residency/binding、remap/release mechanics。
- 承接现有 static span 生命周期：启动 token-only spans；layer-major prefill 按层切换；decode 使用 all-static spans。
- 只消费 raw GGUF bytes；保持默认 Metal mmap-backed 行为，不 eagerly copy 完整 GGUF。
- 不建立 TCIM 私有 policy/orchestration。

## Definition of Done

- [x] model range/span 边界和重映射失败可明确传播。
- [x] token-only → per-layer → all-static 生命周期有可复现 smoke evidence。
- [x] H2D/residency/binding 不读取 span 外字节，不产生 stale binding。
- [x] release/remap 后资源和绑定状态正确。
- [x] 无 packed/reordered/requantized derivative，也无完整 GGUF eager copy。

## Non-goals

- routed expert selected-load、top-6、eviction/reuse seam；这些按 #1 进入 Step 2。
- full-GGUF P2 correctness。
- compute primitive。

**Comment by kkongling** [comment]
#7 验收通过，实现已提交到 `npu/m50`：[commit 95c9659](https://github.com/fixedpointworks/ds4/commit/95c96595eab434cb739dd493d9c5062f9b6bd6d7)。

实现复用 shared GGUF loader/span policy，在 TCIM adapter 内完成 raw-span H2D、residency、lookup binding 和 remap/release。所有边界在扰动旧映射前验证；重映射先撤销旧绑定，只有全部 H2D 成功后才发布新一代绑定；失败释放可重试，不确定完成的传输保持隔离。未引入完整 GGUF eager copy 或 packed/reordered/requantized derivative。`AGENT.md` 同时记录用户对 hq50 的持续传输授权。

验收证据（2026-09-05）：

| 验收项 | 结果 |
|---|---|
| range/span 边界、remap 失败传播 | fake-HAL 覆盖 invalid/overflow/active batch、部分 allocation、release retry、H2D failure/quarantine；真实 HAL 覆盖边界、batch 拒绝及 host 注入的第二次 allocation 失败后恢复，全部通过 |
| token-only → per-layer → all-static | hq50 真 GGUF shared helpers：token-only → prefill layer 0 → 1 → 42 → all-static，全部通过 |
| H2D 与 binding 范围正确、无 stale binding | 每次 source/bytes 与 shared spans 精确一致，全部 H2D 字节经 D2H 逐字节比较；iomap/offset、gap、旧 token binding、all-static 的 routed-expert 不可见性通过 |
| release/remap 后资源正确 | 三轮真实 HAL adapter lifecycle、GGUF cleanup/reinit 与 borrowed model fd 检查通过；最终无残留测试进程，DDR free 从 24448 MB 恢复至 24448 MB |
| raw GGUF，无 full eager copy 或 derivative | 只读 mmap/shared loader，adapter 仅复制明确 spans；observer 验证原始字节，没有生成权重文件 |

设备为 hq50 logical device 0，HMSW/driver V1.4.0。输入是既有 `DeepSeek-V4-Flash-IQ2XXS-w2Q2K-AProjQ8-SExpQ8-OutQ8-chat-v2-imatrix.gguf`：86,720,111,488 字节、1328 tensors、43 层。5 个阶段累计核对 16,832,081,124 字节 H2D/D2H；all-static 使用全部 43 层的 90 spans，共 8,801,384,796 字节（8.197 GiB）。测试前本地/hq50 生产源码与刷新后的 kernel hex 哈希逐项一致。

已执行命令：

```sh
# 本地 fake HAL
bash misc/tcim-runtime-checks/run_adapter_checks.sh host

# hq50，工作目录 /home/sky/ds4，设备测试串行执行
make test-tcim
bash misc/tcim-runtime-checks/run_adapter_checks.sh live
bash misc/tcim-runtime-checks/run_static_gguf.sh --build-only
misc/tcim-runtime-checks/build/test_tcim_static_gguf <raw-GGUF-path>
```

`make test-tcim` 的 CLI/backend selector、logical device/auto-budget、ELF/link provider isolation 全部通过。真实 adapter 三轮和真 GGUF smoke 均 exit=0。详细测试脚本与日志保留在 **ignored `misc/tcim-runtime-checks/`**，本地结果目录为 `issue7-2026-09-05/`（`README.md`、`results.json`、`hq50-build-cli.log`、`hq50-adapter-live.log`、`hq50-gguf-live.log`、源码/二进制哈希）；本提交仅包含 `ds4_tcim.c` 和 `AGENT.md`。将 runtime/static smoke 汇入正式聚合 target 仍由 #2 完成。

范围与限制：per-layer storage smoke 选层 0/1/42，all-static 覆盖全部 43 层；未运行完整 prefill/decode 计算、P2 logits 或 Step 2 selected-expert cache。真实硬件未故意注入 H2D timeout/free failure，此类路径由 fake HAL 验证。SSHFS 约 4.5 MB/s 的单路读取通过有界并发 raw-span 预读加速，最多 32 MiB 临时读缓冲、无权重文件输出，因此耗时不用于性能结论。共享 `ds4.c`、Metal/CUDA/ROCm/distributed 实现未由本变更修改，本轮未做这些后端的实机回归。

结论：#7 五项 Definition of Done 全部满足，按 completed 关闭。Step 1 最终聚合验收继续由 #2 跟进，本次关闭不代表完整 TCIM 推理支持。
